### DISCUSSION TITLE
passlib seems not being maintenanced anymore. However FastAPI's docs still using. Consider change it.

### DISCUSSION THREAD

**Discussion by dann2333** [fastapi/fastapi#11773] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from passlib.context import CryptContext

pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
hashed_password = pwd_context.hash(user.password)
```

### Description

It won't cause a actually problem but a warning about cannot get bcrypt's version(just for logging). 
`(trapped) error reading bcrypt version` 
There are also other issues about it. (https://github.com/pyca/bcrypt/issues/684  https://github.com/langflow-ai/langflow/issues/1173 and more) It seems that passlib has been abandoned.
![4fbe898990913e9e4203cff4fbc481bd](https://github.com/tiangolo/fastapi/assets/146356127/6dfa5e7c-c622-4633-ac6a-1f5ada86e907)

I suggest that the official docs may need to change some contexts such as this https://fastapi.tiangolo.com/tutorial/security/oauth2-jwt/ to avoid the using of passlib since it haven't been maintenanced for a long time.

Thanks.

### Operating System

Windows

### FastAPI Version

0.111.0

### Pydantic Version

2.7.4

### Python Version

Python 3.10.2

**Comment by AIdjis** [comment]
please check this :https://github.com/canonical/cloud-init/issues/4791

**Reply by dann2333** [reply]
Hi, it's 3 months passed and the new version of passlib seems still not released.

**Reply by RuoCJ** [reply]
> please check this :[canonical/cloud-init#4791](https://github.com/canonical/cloud-init/issues/4791) [canonical/cloud-init#4791](https://github.com/canonical/cloud-init/issues/4791)

Hi, passlib has not been updated yet, so it is recommended that fastapi update the documentation and abandon passlib for now

**Reply by dann2333** [reply]
> please check this :[canonical/cloud-init#4791](https://github.com/canonical/cloud-init/issues/4791)

fastapi seems really need to change some... passlib still no update...

**Reply by sinisaos** [reply, marked as answer]
@dann2333 Required changes to the docs example to use [bcrypt](https://pypi.org/project/bcrypt/) directly are minimal. Just need to replace this part of the code 
```python
pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")

oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")

app = FastAPI()


def verify_password(plain_password, hashed_password):
    return pwd_context.verify(plain_password, hashed_password)


def get_password_hash(password):
    return pwd_context.hash(password)
```
with this
```python
# from passlib.context import CryptContext <- remove passlib from imports
import bcrypt

oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")

app = FastAPI()


def verify_password(plain_password, hashed_password):
    return bcrypt.checkpw(
        bytes(plain_password, encoding="utf-8"),
        bytes(hashed_password, encoding="utf-8"),
    )


def get_password_hash(password):
    return bcrypt.hashpw(
        bytes(password, encoding="utf-8"),
        bcrypt.gensalt(),
    )

```
Sorry if I missed your point.

**Comment by suspiciousRaccoon** [comment]
`passlib` seems to be [dead](https://foss.heptapod.net/python-libs/passlib/-/issues/187)
However, there is a new library called [`pwdlib`](https://github.com/frankie567/pwdlib) and someone also made a [fork](https://github.com/ThirVondukr/passlib)
I think the docs should be updated to either use `pwdlib`, or point at the fork (and maybe clarify that `passlib` is the old, unmaintained version)

**Comment by JiaGengChang** [comment]
Came here from the OAuth2 chapter from the docs
The code provided by the docs is broken because passlib CryptContext has stopped working. 
I get `ValueError: password cannot be longer than 72 bytes, truncate manually if necessary (e.g. my_password[:72])` even when trying `passlib.CryptContext.dummy_verify()`
It is high time to replace passlib with the accepted solution in the documentation.
I'm on Python 3.13, passlib 1.7.4, bcrypt 5.0.0

**Reply by JiaGengChang** [reply]
Here's a small example to get the error:
```python
from passlib.context import CryptContext
CryptContext(schemes=["bcrypt"],deprecated="auto").dummy_verify()
```

**Reply by Hadatko** [reply]
Same for me. In debug it behaves well but on server it started to show this error message

A bit more context from bcrypt 5.0.0
<img width="944" height="295" alt="image" src="https://github.com/user-attachments/assets/b8f04a60-88ca-4bbb-ad2f-58fb5904dd4b" />

**Reply by YuriiMotov** [reply]
We have updated docs to use `pwdlib` in code examples instead of `passlib`.
Please, take a look: https://fastapi.tiangolo.com/tutorial/security/oauth2-jwt/#hash-and-verify-the-passwords

**Reply by JiaGengChang** [reply]
Thanks @YuriiMotov ! I saw the update to the docs two days ago and immediately adopted it into my codebase

**Comment by Youjin1985** [comment]
~~updated docs still lead to a problem , this is not working:~~
```
from pwdlib import PasswordHash
password_hash = PasswordHash.recommended()
def get_password_hash(password):
    return password_hash.hash(password)
```
call of get_password_hash('secret')
will get:
    `TypeError: PasswordHasher.hash() got an unexpected keyword argument 'salt'`
error with argon2-21.3.0
works with argon2-23.1.0
